### PR TITLE
[Merged by Bors] - chore: rewrite lakefile.lean

### PR DESCRIPTION
**Pull request by kim-em**
Reorganize lakefile.lean, and remove unnecessary stuff.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by tydeu** [comment]
@semorrison It appears the some of the Archive lean files are making use of `autoImplicit` which is why the build is failing.

File: lakefile.lean [hunk 1]
```diff
@@ -24,43 +24,67 @@ def weakLeanArgs : Array String :=
...
+  supportInterpreter := true
+
+/-- `lake exe checkYaml` verifies that all declarations referred to in `docs/*.yaml` files exist. -/
+lean_exe checkYaml where
+  srcDir := "scripts"
+  supportInterpreter := true
+
+/-- `lake exe graph` constructs import graphs in `.dot` or graphical formats. -/
+lean_exe graph where
+  supportInterpreter := true
```
Comment (digama0) [inline lakefile.lean:82]: Why doesn't this need ``root := `ImportGraph.Main``?

Comment (semorrison) [reply lakefile.lean:82]: Fixed.

**Review by tydeu — APPROVED** [review]

**Comment by kmill** [comment]
This conflicts with #8461, so

bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**